typed_python/StringType.cpp

```diff
@@ -136,7 +222,7 @@ int64_t String::bytesPerCodepointRequiredForUtf8(const uint8_t* utf8Str, int64_t
         if (utf8Str[0] >> 4 == 0b1110) {
             length -= 1;
             utf8Str += 3;
-            bytes_per_codepoint = std::max<int64_t>(4, bytes_per_codepoint);
+            bytes_per_codepoint = std::max<int64_t>(2, bytes_per_codepoint);
```

Comment:
this looks right, but can we add some tests to make sure we're covering this case correctly?